app/helpers/markup_helper.rb

```diff
@@ -0,0 +1,10 @@
+module MarkupHelper
+  def editor_paths
+    {
+      form_url: form_fields_path(format: :js),
+      help_url: help_markup_path,
+      preview_url: preview_markup_path,
+      source_url: source_fields_path
+    }
```

Comment:
What if instead of setting `allow_dropdown` on the form, and passing it through the request to the render we change the url of the request slightly to represent a new vs existing record in a restful-esque manner. Then in the controller based on whether we sent an id or now we set `@allow_dropdown` and render accordingly? /cc @etdsoft. It's pretty much the same concept we're doing now we're just not creating a specific data attribute for it.

```suggestion
  def editor_paths(resource)
    {
      form_url: form_fields_path(resource, format: :js),
      help_url: help_markup_path,
      preview_url: preview_markup_path,
      source_url: source_fields_path
    }
 end
```

Reply:
Unfortunately this doesn't work, specifically for the case when a user switches from Source -> Fields. We're defining the `form_url` to also have the `allow_dropdown` param so every request we make to render the Fields view will have that same param. This doesn't work for the Source -> Fields because we need to pass a `false` `allow_dropdown` here even if the resource is a new record.